src/wp-admin/includes/file.php

```diff
@@ -1946,6 +1946,75 @@ function copy_dir( $from, $to, $skip_list = array() ) {
 	return true;
 }
 
+/**
+ * Moves a directory from one location to another.
+ *
+ * Recursively invalidates OPcache on success.
+ *
+ * If the renaming failed, falls back to copy_dir().
+ *
+ * Assumes that WP_Filesystem() has already been called and setup.
+ *
+ * @since 6.2.0
+ *
+ * @global WP_Filesystem_Base $wp_filesystem WordPress filesystem subclass.
+ *
+ * @param string $from      Source directory.
+ * @param string $to        Destination directory.
+ * @param bool   $overwrite Optional. Whether to overwrite the destination directory if it exists.
+ *                          Default false.
+ * @return true|WP_Error True on success, WP_Error on failure.
+ */
+function move_dir( $from, $to, $overwrite = false ) {
+	global $wp_filesystem;
+
+	if ( trailingslashit( strtolower( $from ) ) === trailingslashit( strtolower( $to ) ) ) {
+		return new WP_Error(
+			'source_destination_same_move_dir',
+			__( 'The source and destination are the same.' )
+		);
+	}
+
+	if ( ! $overwrite && $wp_filesystem->exists( $to ) ) {
+		return new WP_Error( 'destination_already_exists_move_dir', __( 'The destination folder already exists.' ), $to );
+	}
+
+	if ( $wp_filesystem->move( $from, $to, $overwrite ) ) {
```

Comment:
I'm sure this has been tested already, but just double checking: There is no problem when calling this on one of the other filesystem classes that _do not_ have explicit directory support in their `move()` method?

I may have missed it, but is there unit test coverage for that? We should probably cover that this method also works well with the other filesystem classes, if that is at all possible.

Reply:
All other filesystems have explicit directory support in `::move()`.

Reply:
@afragen In that case, can we modify the doc blocks for these other filesystem classes `move()` methods as well, to align with the updated doc block in the direct filesystem class? I think we should make sure they all are aligned and clarify that they _also_ work for directories.

cc @costdev

Reply:
> to align with the updated doc block in the direct filesystem class

+1. Think it only needs a copy/paste of the changes: https://github.com/WordPress/wordpress-develop/pull/3909/files#diff-9d18f7825232295f8adaf0fd5755ee1897d474edff1ef8e804bba531712b9e09R319-R326.

Reply:
I have opened this ticket to add update the docblocks of the other `::move()` methods: https://core.trac.wordpress.org/ticket/57604